### PR TITLE
Enchance release process template with Pre Release

### PR DESCRIPTION
**Pull request by phisco**
Porting of the original commit from @turkenh:
https://github.com/crossplane/crossplane/pull/4871

### CODE REVIEW

**Comment by phisco** [comment]
We really need to convert these steps to dumb proof `gh` commands 😅

**Review by jbw976 — APPROVED** [review]
Cool, I had approved the previous PR, so approving here as well!

One piece of feedback that looks to not have been incorporated was https://github.com/crossplane/crossplane/pull/4871#discussion_r1370673793 - consider that one as well before merging, but its not blocking!

edit: to reduce confusion, the comment i'm pointing to is about splitting into different sections, e.g. for code freeze and for release

**Comment by turkenh** [comment]
Thanks for the PR @phisco! That PR fell through the cracks  🤦 

> One piece of feedback that looks to not have been incorporated was [crossplane/crossplane#4871 (comment)](https://github.com/crossplane/crossplane/pull/4871#discussion_r1370673793) - consider that one as well before merging, but its not blocking!

I was actually waiting for https://github.com/crossplane/crossplane/issues/4875 to be finalized and was planning to take the latest issue description there (e.g. edit and get raw markdown) and update my PR with it. @phisco feel free to do that, or leave this PR to me so that I can push a commit here on monday (assuming this can wait till then).

**Comment by phisco** [comment]
Sure @turkenh, feel free to reopen a pr with it on Monday. Keeping this open just as a memo 👌

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -18,32 +18,51 @@ Please assign the release manager to the issue.
 
 This issue can be closed when we have completed the following steps (in order).
 Please ensure all artifacts (PRs, workflow runs, Tweets, etc) are linked from
-this issue for posterity. Refer to this [prior release issue][release-1.11.0] for
+this issue for posterity. Refer to this [prior release issue][release-1.14.0] for
 examples of each step, assuming release vX.Y.0 is being cut.
 
-- [ ] **[In Crossplane Runtime]**: Prepared the release branch `release-X.Y` at the beginning of [Code Freeze]:
+### Code Freeze
```
Comment (phisco) [inline .github/ISSUE_TEMPLATE/release.md:24]: What about feature freeze? 🤔

Comment (turkenh) [reply .github/ISSUE_TEMPLATE/release.md:24]: I don't think we have steps to perform on feature freeze day. We just stop accepting PRs with features.
Not sure if it is worth adding something here.

**Review by turkenh — APPROVED** [review]
Thank you @phisco 🙌

**Pull request merged**